axelor-account/src/main/java/com/axelor/apps/account/service/fixedasset/FixedAssetServiceImpl.java

```diff
@@ -38,58 +46,239 @@
+        .contains(FixedAssetRepository.DEPRECIATION_PLAN_IFRS)) {
+      FixedAssetLineComputationService fixedAssetLineComputationService =
+          fixedAssetLineServiceFactory.getFixedAssetComputationService(
+              FixedAssetLineRepository.TYPE_SELECT_IFRS);
+      FixedAssetLine initialFiscalFixedAssetLine =
+          fixedAssetLineComputationService.computeInitialPlannedFixedAssetLine(fixedAsset);
+      fixedAsset.addIfrsFixedAssetLineListItem(initialFiscalFixedAssetLine);
+
+      generateComputedPlannedFixedAssetLine(
+          fixedAsset,
+          initialFiscalFixedAssetLine,
+          fixedAsset.getIfrsFixedAssetLineList(),
+          FixedAssetLineRepository.TYPE_SELECT_IFRS);
+    }
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @throws AxelorException
+   * @throws NullPointerException if fixedAsset is null
+   */
+  @Override
+  public void generateAndComputeFiscalFixedAssetLines(FixedAsset fixedAsset)
+      throws AxelorException {
+    Objects.requireNonNull(fixedAsset, ARG_FIXED_ASSET_NPE_MSG);
+    if (fixedAsset
+        .getDepreciationPlanSelect()
+        .contains(FixedAssetRepository.DEPRECIATION_PLAN_FISCAL)) {
+      FixedAssetLineComputationService fixedAssetLineComputationService =
+          fixedAssetLineServiceFactory.getFixedAssetComputationService(
+              FixedAssetLineRepository.TYPE_SELECT_FISCAL);
+      FixedAssetLine initialFiscalFixedAssetLine =
+          fixedAssetLineComputationService.computeInitialPlannedFixedAssetLine(fixedAsset);
+      fixedAsset.addFiscalFixedAssetLineListItem(initialFiscalFixedAssetLine);
+
+      generateComputedPlannedFixedAssetLine(
+          fixedAsset,
+          initialFiscalFixedAssetLine,
+          fixedAsset.getFiscalFixedAssetLineList(),
+          FixedAssetLineRepository.TYPE_SELECT_FISCAL);
+    }
+  }
+  /**
+   * {@inheritDoc}
+   *
+   * @throws AxelorException
+   * @throws NullPointerException if fixedAsset is null
+   */
+  @Override
+  public void generateAndComputeFixedAssetLines(FixedAsset fixedAsset) throws AxelorException {
+    Objects.requireNonNull(fixedAsset, ARG_FIXED_ASSET_NPE_MSG);
+    if (fixedAsset
+        .getDepreciationPlanSelect()
+        .contains(FixedAssetRepository.DEPRECIATION_PLAN_ECONOMIC)) {
+      FixedAssetLineComputationService fixedAssetLineComputationService =
+          fixedAssetLineServiceFactory.getFixedAssetComputationService(
+              FixedAssetLineRepository.TYPE_SELECT_ECONOMIC);
+      FixedAssetLine initialFixedAssetLine =
+          fixedAssetLineComputationService.computeInitialPlannedFixedAssetLine(fixedAsset);
+      fixedAsset.addFixedAssetLineListItem(initialFixedAssetLine);
+
+      generateComputedPlannedFixedAssetLine(
+          fixedAsset,
+          initialFixedAssetLine,
+          fixedAsset.getFixedAssetLineList(),
+          FixedAssetLineRepository.TYPE_SELECT_ECONOMIC);
+    }
+  }
+
+  private List<FixedAssetLine> generateComputedPlannedFixedAssetLine(
```

Comment:
better to rename it generateComputedPlannedFixedAssetLine**s** since it returns a list now

Reply:
done